micrometer-meter-provider/src/main/java/io/opentelemetry/contrib/metrics/micrometer/internal/state/MeterSharedState.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.metrics.micrometer.internal.state;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.opentelemetry.contrib.metrics.micrometer.CallbackRegistration;
+import javax.annotation.Nullable;
+
+/**
+ * State for a meter.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public final class MeterSharedState {
+  private final MeterProviderSharedState providerSharedState;
+  private final String instrumentationScopeName;
+  @Nullable private final String instrumentationScopeVersion;
+  @Nullable private final String schemaUrl;
+
+  public MeterSharedState(
+      MeterProviderSharedState providerSharedState,
+      String instrumentationScopeName,
+      @Nullable String instrumentationScopeVersion,
+      @Nullable String schemaUrl) {
+    this.providerSharedState = providerSharedState;
+    this.instrumentationScopeName = instrumentationScopeName;
+    this.instrumentationScopeVersion = instrumentationScopeVersion;
+    this.schemaUrl = schemaUrl;
+  }
+
+  public MeterRegistry meterRegistry() {
+    return providerSharedState.meterRegistry();
+  }
+
+  public String instrumentationScopeName() {
+    return instrumentationScopeName;
+  }
+
+  @Nullable
+  public String instrumentationScopeVersion() {
+    return instrumentationScopeVersion;
+  }
+
+  @Nullable
+  public String schemaUrl() {
+    return schemaUrl;
+  }
```

Comment:
In OTel metrics API, `Meter`s with different instrumentation scopes are not supposed to influence each other; meaning that they can contain instruments with same names/attributes and they won't be merged together.
WDYT about adding the scope info as tags to all Micrometer instruments created within an OTel `Meter`?